torch/fx/graph.py

```diff
@@ -1801,6 +1803,18 @@ def has_side_effect(node):
                 self.erase_node(node)
                 changed = True
 
+        # Call DCE on the subgraphs
+        if self.owning_module is not None:
+            subgraph_names: OrderedSet[str] = OrderedSet()
+            for node in self.find_nodes(op="get_attr"):
+                attr_name = node.target
+                if "." not in attr_name and attr_name not in subgraph_names:
```

Comment:
an alternative to looping over getattrs might just be to loop over `self.owning_module.children()`. I think looping over the getattrs and ignoring attr names with a period has the risk of skipping nested submodules. In practice this probably doesn't matter though (since under compile, I don't think we'll ever see nested submodules grabbed direct with `getattr` from the top-layer fx graph).

Reply:
Thanks, the refactored looks way cleaner.